spring-cloud-dataflow-shell/src/main/java/org/springframework/cloud/dataflow/shell/command/FieldValueCounterCommands.java

```diff
@@ -0,0 +1,117 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.dataflow.shell.command;
+
+import java.text.NumberFormat;
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cloud.dataflow.rest.client.FieldValueCounterOperations;
+import org.springframework.cloud.dataflow.rest.resource.FieldValueCounterResource;
+import org.springframework.cloud.dataflow.rest.resource.MetricResource;
+import org.springframework.cloud.dataflow.shell.config.DataFlowShell;
+import org.springframework.cloud.dataflow.shell.converter.NumberFormatConverter;
+import org.springframework.hateoas.PagedResources;
+import org.springframework.shell.core.CommandMarker;
+import org.springframework.shell.core.annotation.CliAvailabilityIndicator;
+import org.springframework.shell.core.annotation.CliCommand;
+import org.springframework.shell.core.annotation.CliOption;
+import org.springframework.shell.table.BeanListTableModel;
+import org.springframework.shell.table.CellMatchers;
+import org.springframework.shell.table.Formatter;
+import org.springframework.shell.table.SimpleHorizontalAligner;
+import org.springframework.shell.table.Table;
+import org.springframework.shell.table.TableBuilder;
+import org.springframework.shell.table.TableModel;
+import org.springframework.stereotype.Component;
+
+/**
+ * Commands for interacting with Field Value Counter analytics.
+ * 
+ * @author Eric Bottard
+ */
+@Component
+public class FieldValueCounterCommands extends AbstractMetricsCommands implements CommandMarker {
+
+	protected FieldValueCounterCommands() {
+		super("Field Value Counter");
+	}
+
+	private static final String DISPLAY_COUNTER = "field-value-counter display";
+
+	private static final String LIST_COUNTERS = "field-value-counter list";
+
+	private static final String DELETE_COUNTER = "field-value-counter reset";
+
+	@Autowired
+	private DataFlowShell dataFlowShell;
+
+	@CliAvailabilityIndicator({ LIST_COUNTERS, DISPLAY_COUNTER, DELETE_COUNTER })
+	public boolean available() {
+		return dataFlowShell.getDataFlowOperations() != null;
+	}
+
+	@CliCommand(value = DISPLAY_COUNTER, help = "Display the value of a field value counter")
+	public List<Object> display(
+			@CliOption(key = { "", "name" }, help = "the name of the field value counter to display", mandatory = true
+					/*,optionContext = "existing-counter disable-string-converter"*/) String name,
+			final @CliOption(key = "pattern", help = "the pattern used to format the values (see DecimalFormat)",
+					mandatory = false, unspecifiedDefaultValue = NumberFormatConverter.DEFAULT) NumberFormat pattern) {
+		FieldValueCounterResource counter = fvcOperations().retrieve(name);
+
+		LinkedHashMap<String, Object> header = new LinkedHashMap<>();
+		header.put("key", "Key");
```

Comment:
I'm not sure, because it's quite different from the fieldName that was used when creating the counter. This is an actual value that *that* fieldName was seen with, while `Value` on the line below is the number of times it was seen with that value.

I think we could do `Key -> Value, Value -> Count`